### PR TITLE
Zebras timeframe

### PR DESCRIPTION
**Pull request by jaheba**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Please tag this pr with at least one of these labels to make our release process faster:** BREAKING, new feature, bug fix, other change, dev setup

### CODE REVIEW

File: src/gluonts/zebras/_timeframe.py [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+
+    return dataclasses.replace(clone, **kwargs)
+
+
+@dataclasses.dataclass
+class TimeFrame:
+    columns: dict
+    index: Optional[Periods]
+    static: dict
+    length: int
```
Comment (lostella) [inline src/gluonts/zebras/_timeframe.py:101]: Could this be a `@property` instead?

File: src/gluonts/zebras/_timeframe.py [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+
+
+def time_frame(
+    columns=None,
+    index=None,
+    start=None,
+    freq=None,
+    static=None,
+    tdims=None,
+    length=None,
```
Comment (lostella) [inline src/gluonts/zebras/_timeframe.py:536]: I'm wondering when would one need to specify this?

File: src/gluonts/zebras/_timeframe.py (hunk 1 above)
Comment (jaheba) [reply src/gluonts/zebras/_timeframe.py:101]: No, because `columns` could be empty.

File: src/gluonts/zebras/_timeframe.py (hunk 2 above)
Comment (jaheba) [reply src/gluonts/zebras/_timeframe.py:536]: If you create an empty instance which you want to build up.

Comment (lostella) [reply src/gluonts/zebras/_timeframe.py:536]: Wouldn't one in that case provide a non-empty set of columns, each one being empty?

Comment (jaheba) [reply src/gluonts/zebras/_timeframe.py:536]: Not sure what you mean, but what happens if you drop all columns -- the length should stay the same.

**Review by lostella — APPROVED** [review]

**Pull request merged**